services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/crt/S3CrtAsyncHttpClient.java

```diff
@@ -122,6 +132,33 @@ private static S3MetaRequestOptions.MetaRequestType requestType(AsyncExecuteRequ
         return S3MetaRequestOptions.MetaRequestType.DEFAULT;
     }
 
+    private ChecksumAlgorithm crtChecksumAlgorithm(AsyncExecuteRequest asyncRequest) {
+        HttpChecksum httpChecksum = asyncRequest.httpExecutionAttributes().getAttribute(HTTP_CHECKSUM);
+
+        if (checksumNotApplicable(httpChecksum)) {
+            return null;
+        }
+
+        // TODO: revisit default checksum
+        Algorithm algorithm = httpChecksum.requestAlgorithm() == null ? Algorithm.CRC32 :
```

Comment:
What are the results of running different algorithms?

Reply:
From my previous tests, I did not notice much difference, but it may be because the machine I use have SHA acceleration. I'd need to run more tests.